### PR TITLE
[Merged by Bors] - feat(Topology/Algebra/NormedValued): add correspondence

### PR DESCRIPTION
**Pull request by mariainesdff**
Nontrivial nonarchimedean norms correspond to rank one valuations.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Algebra/NormedValued.lean [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+end NormedField
+
+namespace Valued
+
+variable {L : Type*} [Field L] {Γ₀ : Type*} [LinearOrderedCommGroupWithZero Γ₀]
+  [val : Valued L Γ₀] [hv : RankOne val.v]
+
+/-- The norm function determined by a rank one valuation on a field `L`. -/
+def norm_def : L → ℝ := fun x : L => hv.hom (Valued.v x)
```
Comment (kbuzzard) [inline Mathlib/Topology/Algebra/NormedValued.lean:67]: `norm_def` sounds more like the name of a lemma saying that norm = something it's defined to be. Is there a reason that this isn't just called `norm`?

File: Mathlib/Data/Real/NNReal.lean [hunk 2]
```diff
@@ -1193,6 +1193,42 @@ theorem cast_natAbs_eq_nnabs_cast (n : ℤ) : (n.natAbs : ℝ≥0) = nnabs n :=
...
 
+section StrictMono
+
+open NNReal
+
+variable {Γ₀ : Type*} [LinearOrderedCommGroupWithZero Γ₀]
+
+/-- If `Γ₀ˣ` is nontrivial and `f : Γ₀ →*₀ ℝ≥0` is a strict monomorphism, then for any positive
+  `r : ℝ≥0`, there exists `d : Γ₀ˣ` with `f d < r`. -/
+theorem NNReal.exists_strictMono_lt [h : Nontrivial Γ₀ˣ] {f : Γ₀ →*₀ ℝ≥0} (hf : StrictMono f)
```
Comment (kbuzzard) [inline Mathlib/Data/Real/NNReal.lean:1204]: I feel like we can do a bit better with the name of this: e.g. `NNReal.exists_lt_of_strictMono` (and similar comment for the theorem below).

File: Mathlib/Topology/Algebra/NormedValued.lean [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+          have hδ_pos : 0 < δ := by
+            rw [hδ, ← _root_.map_zero hv.hom]
+            exact hv.strictMono (Units.zero_lt ε)
+          use δ, hδ_pos
+          apply subset_trans _ hε
+          intro x hx
+          simp only [mem_setOf_eq, norm_def, hδ, NNReal.val_eq_coe, NNReal.coe_lt_coe] at hx
+          rw [mem_setOf, ← neg_sub, Valuation.map_neg]
+          exact (RankOne.strictMono Valued.v).lt_iff_lt.mp hx
+        · letI : Nontrivial Γ₀ˣ := (nontrivial_iff_exists_ne (1 : Γ₀ˣ)).mpr
```
Comment (kbuzzard) [inline Mathlib/Topology/Algebra/NormedValued.lean:114]:
```suggestion
        · haveI : Nontrivial Γ₀ˣ := (nontrivial_iff_exists_ne (1 : Γ₀ˣ)).mpr
```
(it's a Prop)

File: Mathlib/Topology/Algebra/NormedValued.lean [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        · set δ : ℝ≥0 := hv.hom ε with hδ
+          have hδ_pos : 0 < δ := by
+            rw [hδ, ← _root_.map_zero hv.hom]
+            exact hv.strictMono (Units.zero_lt ε)
+          use δ, hδ_pos
+          apply subset_trans _ hε
+          intro x hx
+          simp only [mem_setOf_eq, norm_def, hδ, NNReal.val_eq_coe, NNReal.coe_lt_coe] at hx
+          rw [mem_setOf, ← neg_sub, Valuation.map_neg]
+          exact (RankOne.strictMono Valued.v).lt_iff_lt.mp hx
```
Comment (kbuzzard) [inline Mathlib/Topology/Algebra/NormedValued.lean:113]: These lines look very similar to 118-123, which I think are proving the other direction. Can they be factored out and can you prove both directions at once, or is this more trouble than it's worth?

**Review by kbuzzard — COMMENTED** [review]
This looks almost ready to go. I've left some comments.

File: Mathlib/Topology/Algebra/NormedValued.lean (hunk 1 above)
Comment (mariainesdff) [reply Mathlib/Topology/Algebra/NormedValued.lean:67]: I can rename it to `norm`.

File: Mathlib/Topology/Algebra/NormedValued.lean (hunk 4 above)
Comment (mariainesdff) [reply Mathlib/Topology/Algebra/NormedValued.lean:113]: Yes, each of these blocks proves one of the inclusions. I don't see a quick way of unifying the proofs.

**Review by adamtopaz — COMMENTED** [review]
This looks good to me! The one issue I see is around naming. Edit: ignore my last comment -- I looked more closely and now understand what is being done here. IMO the property of a valuation being rank one is actually a `Prop` (some property about the type of convex subgroups of the value group), but now it's defined in a way that includes the data of the embedding into the reals. We can think about doing something about this later on.

Anyway, there is one minor comment below, then this PR should be good to go!

**Comment by adamtopaz** [comment]
bors d+

**Pull request closed without merging**